src/containers/withAuthorizedAction/withAuthorizedAction.tsx

```diff
@@ -0,0 +1,163 @@
+import { RootStateOrAny, connect } from 'react-redux'
+import React, { ComponentProps, useCallback, useState } from 'react'
+import { BigNumber, ethers } from 'ethers'
+import {
+  Authorization,
+  AuthorizationType
+} from '../../modules/authorization/types'
+import { getNetworkProvider } from '../../lib/eth'
+import { getAddress } from '../../modules/wallet/selectors'
+import {
+  AuthorizationModal,
+  AuthorizationStepStatus,
+  AuthorizedAction
+} from './AuthorizationModal'
+import {
+  WithAuthorizedActionProps,
+  MapStateProps,
+  AuthorizeActionOptions
+} from './withAuthorizedAction.types'
+import { getERC20ContractInstance, getERC721ContractInstance } from './utils'
+
+const mapState = (state: RootStateOrAny): MapStateProps => ({
```

Comment:
should we move this logic to a `withAuthorizedAction.container.tsx` file or pass it as a required prop in `WithAuthorizedActionProps `?

Reply:
This is not data that withAuthorizedAction is receiving, but instead part of the wrapper component we are returning in the HOC